### PR TITLE
EOS-20327: Audit logs sorting on text fields

### PR DESCRIPTION
**Pull request by tb-s**
# Backend

## Problem Statement

[EOS-20327: Audit logs sorting on text fields](https://jts.seagate.com/browse/EOS-20327)
[EOS-20487: Remove restrictions on sortable fields](https://jts.seagate.com/browse/EOS-20487)
[EOS-20488: Ensure ElasticSearch mappings for `csmauditlogs` are compatible with CSM business logic](https://jts.seagate.com/browse/EOS-20488)

See ticket for details.

## Unit testing on RPM done

Yes

## Problem Description

See tickets for details.

### Description

- Force CSM to generate ElasticSearch mappings for `CsmAuditLogModel` entries;
- Remove restrictions on sortable fields---all fields on audit logs are sortable now.

### Known issues

- Descending order by default; we might benefit of different default orderings per field.

## Test Cases

For each field in audit logs, check if they can be sorted in both ascending and descending order.

Query example: `GET /api/v2/auditlogs/show/csm?start_date=0&end_date=1620319233&sortby=response_code&dir=asc`

- `sortby` selects the field; possible fields: `timestamp` (date/time), `user` (string), `remote_ip` (string), `forwarded_for_ip` (string), `method` (string), `path` (string), `user_agent` (string), `response_code` (integer), `request_id` (integer)
- `dir` can be `asc` for ascending order, or `desc` for descending order.

### CODE REVIEW

File: csm/core/agent/api.py
Comment (tb-s) [inline csm/core/agent/api.py:363]: Please have in mind that this will be called for every single API request, which is objectively worse than being called during startup as the original patch proposed.  It will also have a bigger impact on the first request received every time `csm_agent` comes back up.

@kupranay @pawankumarsrivastava, FYI.

Comment (kupranay) [reply csm/core/agent/api.py:363]: we should have a flag which will be false for the first time and get updated to true once db is initialized. 
"await CsmRestApi._initialize_audit_log_db()" should be called only once when csm-agent will get started and flag should be reset

Comment (tb-s) [reply csm/core/agent/api.py:363]: This change is already part of the patch.  Please check https://github.com/Seagate/cortx-manager/pull/462/files#diff-976c77a4ccd126b2da94f02da86d6fbd963c46d3c34ae2f82b9eb52d607a9981R332

**Review by pawankumarsrivastava — APPROVED** [review]

**Comment by tb-s** [comment]
Updated base branch to `t-main`.

**Comment by tb-s** [comment]
@kupranay Hi, any updates re: this PR?

**Comment by tb-s** [comment]
@pawankumarsrivastava @udayan-y21 @soniyamoholkar @kupranay Hi, any updates re: this PR?  Most likely a 2nd positive review is required before merging.  Please acknowledge (exception to @pawankumarsrivastava since he has already reviewed the PR).

File: csm/core/agent/api.py
Comment (udayan-y21) [inline csm/core/agent/api.py:347]: Instead of reloading the database config.. Reuse the already loaded config.
In csm_agent.py, db config is already loaded.
`Conf.load(const.DATABASE_INDEX, f"yaml://{const.DATABASE_CONF}")`

In api.py, you can use it as 
`db_config = Conf.get(const.DATABASE_INDEX, "databases>es_db>config")`

Can you try this out?

**Review by udayan-y21 — CHANGES_REQUESTED** [review]

File: csm/core/agent/api.py
Comment (tb-s) [reply csm/core/agent/api.py:347]: Hi, @udayan-y21, thank you for your suggestion.  I copied this code snippet from `csm/core/agent/csm_agent.py`.  Although I believe that would work, I would rather keep the snippets equal.

Comment (tb-s) [reply csm/core/agent/api.py:347]: @udayan-y21 Hi, please provide your opinion on this.

Comment (kupranay) [reply csm/core/agent/api.py:325]: as per new guidelines we should be using confstore concept and should not be using  any functions like YAML as much as possible. We dont want to have a code change if in future we move from yaml to json or any other format with similar structure

Comment (tb-s) [reply csm/core/agent/api.py:325]: @kupranay  As I have explaned previously, this code snippet is exactly the one used by `csm/core/agent/csm_agent.py`.  Any further changes to it should, at least in theory, be validated in a separate ticket because they were not even introduced to our codebase yet.

Comment (tb-s) [reply csm/core/agent/api.py:325]: @kupranay @udayan-y21 Here is the link to the original implementation: https://github.com/Seagate/cortx-manager/blob/tmain/csm/core/agent/csm_agent.py#L53

Comment (udayan-y21) [reply csm/core/agent/api.py:325]: @tb-s  I agree to pranay's comment. I suggest change needs to be added in both places.. The one you made the change and second in `csm/core/agent/csm_agent.py` . This is how we can avoid further modification in this area..

@kupranay @tb-s  Im aprroving the PR due to sprint timeline constraint. But change needs to be added either now or in future sprint.

**Review by udayan-y21 — APPROVED** [review]
Approving with change request note
https://github.com/Seagate/cortx-manager/pull/462#discussion_r642026258
 

> @tb-s I agree to pranay's comment. I suggest change needs to be added in both places.. The one you made the change and second in csm/core/agent/csm_agent.py . This is how we can avoid further modification in this area..
> @kupranay @tb-s Im aprroving the PR due to sprint timeline constraint. But change needs to be added either now or in future sprint.

File: csm/core/agent/api.py
Comment (tb-s) [reply csm/core/agent/api.py:325]: I agree that this change must be done.  I don't believe, though, that this story is the right place to test and validate it, only because the suggestion has not been validated before.  We are not currently using it.

I can prepare a separate PR with the changes proposed.

**Pull request merged**
